silhouette/app/com/mohiva/play/silhouette/api/services/AuthenticatorService.scala

```diff
@@ -20,9 +20,52 @@
 package com.mohiva.play.silhouette.api.services
 
 import com.mohiva.play.silhouette.api.{ Authenticator, LoginInfo }
+import play.api.libs.concurrent.Execution.Implicits._
 import play.api.mvc.{ Headers, RequestHeader, Result }
 
 import scala.concurrent.Future
+import scala.language.implicitConversions
+import scala.util.Try
+
+/**
+ * A marker result which indicates that an operation on an authenticator was processed and
+ * therefore it shouldn't updated automatically.
+ *
+ * Due the fact that the update method gets called on every subsequent request to update the
+ * authenticator related data in the backing store and in the result, it isn't possible to
+ * discard or renew the authenticator simultaneously. This is because the "update" method would
+ * override the result created by the "renew" or "discard" method, because it will be executed
+ * as last in the chain.
+ *
+ * As example:
+ * If we discard the session in a Silhouette action then it will be removed from session. But
+ * at the end the update method will embed the session again, because it gets called with the
+ * result of the action.
+ *
+ * @param result The result to wrap.
+ */
+class AuthenticatorResult(result: Result) extends Result(result.header, result.body, result.connection)
+
+/**
+ * The companion object.
+ */
+object AuthenticatorResult {
+
+  /**
+   * Converts a [[Result]] into [[AuthenticatorResult]]
+   */
+  implicit def toAuthenticatorResult(result: Result): AuthenticatorResult = apply(result)
+  implicit def toAuthenticatorResultF(result: Future[Result]): Future[AuthenticatorResult] = result.map(apply)
+  implicit def toAuthenticatorResultT(result: Try[Result]): Try[AuthenticatorResult] = result.map(apply)
+
```

Comment:
So looking around the internet I don't think overriding the copy method is an issue.  I think the main  issue is that subclassing a case class is considered bad form. From what I've been able to gather, case classes are meant to be leafs of a class hierarchy so that they can hold the equivalence relation. By introducing a subclass of Result we slightly taint that property...equality is no longer straight forward.

http://stackoverflow.com/a/11159036

Having said that, I think its an acceptable price to pay for this